docs/examples/tutorial_signal_processing.py

```diff
@@ -0,0 +1,286 @@
+NES = nap.TsdFrame(
+    t=data["ElectricalSeries"].index.values,
+    d=data["ElectricalSeries"].values,
+    time_support=data["ElectricalSeries"].time_support,
+)
+wake_minute = NES.value_from(NES, nap.IntervalSet(900, 960))
+sleep_minute = NES.value_from(NES, nap.IntervalSet(0, 60))
+
+# %%
+# ***
+# Plotting the LFP activity of one slices
+# -----------------------------------
+# Let's plot
+fig, ax = plt.subplots(2)
+for channel in range(sleep_minute.shape[1]):
+    ax[0].plot(
+        sleep_minute.index.values,
+        sleep_minute[:, channel],
+        alpha=0.5,
+        label="Sleep Data",
+    )
+ax[0].set_title("Sleep ephys")
+for channel in range(wake_minute.shape[1]):
+    ax[1].plot(
+        wake_minute.index.values, wake_minute[:, channel], alpha=0.5, label="Wake Data"
+    )
+ax[1].set_title("Wake ephys")
+plt.show()
+
+
+# %%
+# There is much shared information between channels, and wake and sleep don't seem visibly different.
+# Let's take the Fourier transforms of one channel for both and see if differences are present
+channel = 1
+fig, ax = plt.subplots(1)
+fft_sig, fft_freqs = nap.compute_spectrum(sleep_minute[:, channel], fs=int(FS))
+ax.plot(fft_freqs, np.abs(fft_sig), alpha=0.5, label="Sleep Data")
+ax.set_xlim((0, FS / 2))
+fft_sig, fft_freqs = nap.compute_spectrum(wake_minute[:, channel], fs=int(FS))
+ax.plot(fft_freqs, np.abs(fft_sig), alpha=0.5, label="Wake Data")
+ax.set_title(f"Fourier Decomposition for channel {channel}")
+ax.legend()
+plt.show()
+
+
+# %%
+# There seems to be some differences presenting themselves - a bump in higher frequencies for waking data?
+# Let's explore further with a wavelet decomposition
+
+
+def plot_timefrequency(times, freqs, powers, x_ticks=5, y_ticks=5, ax=None, **kwargs):
+    if np.iscomplexobj(powers):
+        powers = abs(powers)
+    ax.imshow(powers, aspect="auto", **kwargs)
+    ax.invert_yaxis()
+    ax.set_xlabel("Time (s)")
+    ax.set_ylabel("Frequency (Hz)")
+    if isinstance(x_ticks, int):
+        x_tick_pos = np.linspace(0, times.size, x_ticks)
+        x_ticks = np.round(np.linspace(times[0], times[-1], x_ticks), 2)
+    else:
+        x_tick_pos = [np.argmin(np.abs(times - val)) for val in x_ticks]
+    ax.set(xticks=x_tick_pos, xticklabels=x_ticks)
+    if isinstance(y_ticks, int):
+        y_ticks_pos = np.linspace(0, freqs.size, y_ticks)
+        y_ticks = np.round(np.linspace(freqs[0], freqs[-1], y_ticks), 2)
+    else:
+        y_ticks_pos = [np.argmin(np.abs(freqs - val)) for val in y_ticks]
+    ax.set(yticks=y_ticks_pos, yticklabels=y_ticks)
+
+
+fig, ax = plt.subplots(2)
+freqs = np.array(
+    [
+        2.59,
+        3.66,
+        5.18,
+        8.0,
+        10.36,
+        14.65,
+        20.72,
+        29.3,
+        41.44,
+        58.59,
+        82.88,
+        117.19,
+        165.75,
+        234.38,
+        331.5,
+        468.75,
+        624.0,
+    ]
+)
+mwt_sleep = nap.compute_wavelet_transform(
```

Comment:
I would explain in more steps here.

Reply:
I'm not sure what I could do without making private functions in signal_processing.py public. Perhaps I could make the Morelet wavelet function public, and plot some wavelets of different scales first? To give users an intuition of what a wavelet is?